kernel/src/grant.rs

```diff
@@ -273,6 +282,15 @@ impl<'a, T: Default> Iterator for Iter<'a, T> {
                 return res;
             }
         }
+        // After running through all real apps, pass the debug app (idx 255) to
+        // the grant iterator in case it has state the capsule needs to process.
+        if self.index == self.len {
+            self.index += 1;
+            let res = self.grant.grant(AppId::new(255));
+            if res.is_some() {
+                return res;
+            }
```

Comment:
I think you can simplify this to `return self.grant.grant(AppId::new(255))` because we anyway return Some or None and because it is last it should be ok?! But maybe better with this `more verbose solution`

Reply:
I tried that, it doesn't work. I think it ends up in an infinite loop.